.github/workflows/ci_linux.yml

```diff
@@ -31,51 +28,74 @@ jobs:
       fail-fast: true
       matrix:
         include:
-          - name: "Unit gcc11"
+          - name: "gcc11"
             cxx: "g++-11"
             cc: "gcc-11"
-            build: unit
             build_type: Release
-            build_threads: 2
-            test_threads: 2
+            cxx_flags: "-std=c++20"
 
-          - name: "Unit gcc10"
+          - name: "gcc10"
             cxx: "g++-10"
             cc: "gcc-10"
-            build: unit
             build_type: Release
-            build_threads: 2
-            test_threads: 2
+            cxx_flags: "-std=c++20"
+
+          - name: "gcc9"
+            cxx: "g++-9"
+            cc: "gcc-9"
+            build_type: Release
+            cxx_flags: "-std=c++2a"
+
+          - name: "clang12"
+            cxx: "clang++-12"
+            cc: "clang-12"
+            build_type: Release
+            cxx_flags: "-std=c++20"
+
+          - name: "clang11"
+            cxx: "clang++-11"
+            cc: "clang-11"
+            build_type: Release
+            cxx_flags: "-std=c++20"
+
+          - name: "clang10"
+            cxx: "clang++-10"
+            cc: "clang-10"
+            build_type: Release
+            cxx_flags: "-std=c++20"
 
     steps:
       - name: Checkout TDL
         uses: actions/checkout@v2
         with:
           path: tdl
-          fetch-depth: 3
-          submodules: recursive
+          fetch-depth: 2
+          submodules: false
 
-      - name: Add package source
-        run: |
-          sudo add-apt-repository --no-update --yes ppa:ubuntu-toolchain-r/ppa
-          sudo add-apt-repository --no-update --yes ppa:ubuntu-toolchain-r/test
-          sudo apt-get update
+      - name: Checkout SeqAn3
+        uses: actions/checkout@v2
+        with:
+          repository: seqan/seqan3
+          ref: 383e8f8d638f8a727a79136ce96ae7642befe39f
+          path: seqan3
+          fetch-depth: 2
+          submodules: false
+
+      - name: Configure APT
+        continue-on-error: true
+        run: bash ./seqan3/.github/workflows/scripts/configure_apt.sh
 
       - name: Install CMake
-        run: bash ./tdl/.github/workflows/scripts/install_cmake.sh
+        run: bash ./seqan3/.github/workflows/scripts/install_cmake.sh
```

Comment:
Hm.. I'm still not sure if that's the best way to do it. 

My thoughts: 
1. If we ever change the location name or content of this file, the CI here directly breaks without a warning and if you and me are not here anymore it might be hard to track down the issue. 
2. I think in general it is not "common practice" to clone an entire repo just to use some scripts. I know it's DRY that way in case we fix something but it still feels a little weird.
3. If we ever do want to change something in this file, we have to copy it over anyway.

Proposal: How about an infrastructure scripts repo? We also re-use a lot of cmake stuff in the app template an will re-use a lot in the argument parser library that should be stand-along (in contrast to the app-template).

Reply:
> Hm.. I'm still not sure if that's the best way to do it. 
> 
> My thoughts: 
> 1. If we ever change the location name or content of this file, the CI here directly breaks without a warning and if you and me are not here anymore it might be hard to track down the issue. 

seqan3 is pinned to a fixed commit

> 2. I think in general it is not "common practice" to clone an entire repo just to use some scripts. I know it's DRY that way in case we fix something but it still feels a little weird.

cloning seqan3 is 1-2 seconds, it's really not that much when its shallow and non-recursive. For me, it's an OK cost to get DRY

> 3. If we ever do want to change something in this file, we have to copy it over anyway.

Yes, however, these are very general scripts, so if you wanted to change something, it might be a fix that should be put into seqan3 anyway...

> 
> Proposal: How about an infrastructure scripts repo? We also re-use a lot of cmake stuff in the app template an will re-use a lot in the argument parser library that should be stand-along (in contrast to the app-template).

In general, yes. The gitHub scripts stuff should be rather easy; all that cmake probably not so much.

Reply:
Fine. I still don't like it :D

Reply:
I don't like it that much either, but it's way better than having the scripts in every repo. Because then you have to copy/paste fixes everywhere instead of just bumping the CI-seqan3-checkout.

Putting the `./github/workflows/scripts` into another repo is easy and something that we can do soon.
Ideally, we would write our own action that does all the CI stuff  - but that needs some time :)